packages/phone-number-privacy/signer/src/signer/io.abstract.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ErrorType,
+  FailureResponse,
+  OdisRequest,
+  OdisResponse,
+  SignerEndpoint,
+  SuccessResponse,
+  WarningMessage,
+} from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
+import { Request, Response } from 'express'
+import { Session } from './action.interface'
+
+export abstract class IOAbstract<R extends OdisRequest> {
+  abstract readonly enabled: boolean
+  abstract readonly endpoint: SignerEndpoint
+
+  abstract init(
+    request: Request<{}, {}, unknown>,
+    response: Response<OdisResponse<R>>
+  ): Promise<Session<R> | null>
```

Comment:
It's unclear what a return value of `null` is from the interface. A comment here and in controller would be helpful. (Doc comments on all these methods would be helpful)